packages/components/src/components/Autocomplete/Autocomplete.tsx

```diff
@@ -0,0 +1,221 @@
+import React, { useEffect, useRef, useState } from "react";
+import { LayoutRectangle, TextInput, View } from "react-native";
+import { ScrollView } from "react-native-gesture-handler";
+import { useStyles } from "../../hooks/useStyles";
+import { EDSStyleSheet } from "../../styling";
+import { IconButton } from "../Button/IconButton";
+import { Input, InputProps } from "../Input";
+import { Menu } from "../Menu";
+
+type ToggleOptionProps = {
+    title: string;
+    active: boolean;
+    iconName: "radiobox-marked" | "radiobox-blank";
+    onPress: () => void;
+};
+
+type AutocompleteProps = {
+    /**
+     * Options list that will be displayed in the autocomplete.
+     */
+    options: string[];
+    /**
+     * A callback method invoked when the user selects an option from the autocomplete.
+     */
+    onSelect: (option: string) => void;
+    /**
+     * A label above the input field.
+     */
+    label?: string;
+    /**
+     * A placeholder for the input field.
+     */
+    placeholder?: string;
+    toggleOptions?: ToggleOptionProps[];
+} & InputProps;
+
+export const Autocomplete: React.FC<AutocompleteProps> = ({
+    options,
+    onSelect,
+    onChange,
+    label,
+    placeholder,
+    toggleOptions,
+    ...restProps
+}) => {
+    const [inputValue, setInputValue] = useState("");
+    const [filteredOptions, setFilteredOptions] = useState(options);
+    const handleMenuOpen = () => {
+        setIsOptionsVisible(true);
+        inputRef.current && inputRef.current.focus();
+    };
+    const handleMenuClose = () => {
+        setIsOptionsVisible(false);
+        inputRef.current && inputRef.current.blur();
+    };
+
+    const [isOptionsVisible, setIsOptionsVisible] = useState(false);
+    const inputRef = useRef<TextInput>(null);
+    const [inputLayout, setInputLayout] = useState<LayoutRectangle | undefined>();
+    const [selectedToggleOptions, setSelectedToggleOptions] = useState<string[]>([]);
+    const styles = useStyles(themedStyles, { inputLayout });
+
+    const handleClearText = () => {
+        setInputValue("");
+        handleMenuClose();
+    };
+    const handleToggleOptionPress = (optionTitle: string) => {
+        if (selectedToggleOptions.includes(optionTitle)) {
+            setSelectedToggleOptions(prev => prev.filter(title => title !== optionTitle));
+        } else {
+            setSelectedToggleOptions(prev => [...prev, optionTitle]);
+        }
+    };
+    const handleClearSelectedOptions = () => {
+        setSelectedToggleOptions([]);
+    };
+
+    useEffect(() => {
+        setFilteredOptions(
+            options.filter(option => option.toLowerCase().includes(inputValue.toLowerCase())),
+        );
+    }, [inputValue]);
```

Comment:
I don't think it would hurt to add `options` as a dependency here, but if you decide not to, I would disable the eslint rule.